### PR TITLE
[Network] Added missing paths to LoadBalancer, NetworkInterface and VirtualNetworkGateway

### PR DESCRIPTION
**Pull request by anton-evseev**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/loadBalancer.json [hunk 1]
```diff
@@ -1059,6 +1481,70 @@
...
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/LoadBalancingRule"
+          },
+          "description": "A list of load balancing rules in a load balancer."
+        },
+        "nextLink": {
+          "type": "string",
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/loadBalancer.json]: Please mark all nextLink properties as readOnly

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json [hunk 2]
```diff
@@ -204,6 +204,48 @@
...
+            "description": "The name of the virtual network gateway."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json]: Is this ```x-ms-pageable```

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/networkInterface.json [hunk 3]
```diff
@@ -324,10 +324,149 @@
...
+            "description": "The name of the network interface."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/networkInterface.json]: nit: is this ```x-ms-pageable```?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Partial review, will add more comments

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/networkInterface.json (hunk 3 above)
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/networkInterface.json]: Yes, updated

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json (hunk 2 above)
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json]: Yes, updated

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/loadBalancer.json (hunk 1 above)
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/loadBalancer.json]: Done

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json (hunk 2 above)
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json]: Yes, updated

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/networkInterface.json (hunk 3 above)
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/networkInterface.json]: Yes, updated

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json [hunk 4]
```diff
@@ -1270,7 +1318,7 @@
       },
       "description": "VirtualNetworkGatewaySku details"
     },
-	"BgpSettings": {
+    "BgpSettings": {
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json]: nit: formatting

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json [hunk 5]
```diff
@@ -1514,39 +1564,54 @@
...
           "description": "The time at which connection was established in Utc format."
         }
       },
       "description": "VirtualNetworkGatewayConnection properties"
     },
+    "GatewayReference": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Resource ID."
```
Comment (dsgouda) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json]: nit: Resource usually refers to a top level ARM resource, please provide a more detailed description.

**Review by dsgouda — CHANGES_REQUESTED** [review]
Other than the one comment, LGTM
Also, after adding nextLink did you have to regenerate the examples?

File: specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json (hunk 5 above)
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/2017-06-01/virtualNetworkGateway.json]: Done

**Comment by number213** [comment]
> after adding nextLink did you have to regenerate the examples?
No, I didn't.

**Comment by dsgouda** [comment]
>>    after adding nextLink did you have to regenerate the examples?
> No, I didn't.
Might be a good idea but I won't block on this

**Review by dsgouda — APPROVED** [review]

**Pull request merged**